halo2_backend/src/plonk/prover.rs

```diff
@@ -563,14 +643,19 @@ impl<
             )
             .collect();
 
-        // Evaluate the h(X) polynomial
+        // 7. Evaluate the h(X) polynomial -----------------------------------------------------------
```

Comment:
I would change this to the following
```
// 7. Given `h(X) = g(X) / t(X)` where:
// - `g(X)` is the gates and arguments polynomial
// - `t(X)` is the vanishing polynomial
// - `h(X)` is the quotient polynomial
// Calculate the g(X) polynomial
```

The reason I say `g(X)` instead of `h(X)` is because if you look at the implementation of `evaluate_h` it actually returns `g`

Reply:
For reference, I took `h` and `t` from the halo2 book https://zcash.github.io/halo2/design/proving-system/vanishing.html
As for `g`, it's not called like that in the halo2 book, it just doesn't have a name.

Then the plonk paper uses different names.